nRFMeshProvision/Classes/Mesh Messages/Generic Messages/GenericCTLStatus.swift

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2019, Nordic Semiconductor
+ * All rights reserved.
+ 
+ * Created by codepgq
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this
+ *    list of conditions and the following disclaimer in the documentation and/or
+ *    other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may
+ *    be used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import Foundation
+
+public struct GenericCTLStatus: GenericMessage, TransitionStatusMessage {
+    
+    public static var opCode: UInt32 = 0x8260
+    public var parameters: Data?
+    
+    public let presentLightness: UInt16
+    public let presentTemperature: UInt16
+    public let targetLightness: UInt16?
+    public let targetTemperature: UInt16?
+    public var remainingTime: TransitionTime?
+    
+    
+    public init?(parameters: Data) {
+        guard parameters.count == 4 || parameters.count == 9 else {
+            return nil
+        }
+        self.presentLightness = UInt16(parameters[0]) | (UInt16(parameters[1]) << 8)
```

Comment:
You may use `parameters.read(offset:)` in all those messages. It's more readable and will not create endiannes problems.